mne/viz/evoked.py

```diff
@@ -1918,20 +1918,20 @@ def plot_compare_evokeds(evokeds, picks=None, gfp=False, colors=None,
             axes = [axes]
         _validate_if_list_of_axes(axes, obligatory_len=len(ch_types))
     else:
-        axes = [plt.subplots(figsize=(8, 6))[1] for _ in range(len(ch_types))]
+        axes = (plt.subplots(figsize=(8, 6))[1] for _ in range(len(ch_types)))
 
     if len(ch_types) > 1:
         logger.info("Multiple channel types selected, returning one figure "
                     "per type.")
         figs = list()
-        for ii, t in enumerate(ch_types):
+        for t, ax in zip(ch_types, axes):
             picks_ = picks_by_types[t]
             title_ = "GFP, " + t if (title is None and gfp is True) else title
             figs.append(plot_compare_evokeds(
                 evokeds, picks=picks_, gfp=gfp, colors=colors,
                 linestyles=linestyles, styles=styles, vlines=vlines, ci=ci,
                 truncate_yaxis=truncate_yaxis, ylim=ylim, invert_y=invert_y,
-                axes=axes[ii], title=title_, show=show))
+                axes=ax, title=title_, show=show))
```

Comment:
this seems to be a purely cosmetic change. What do I miss?

Reply:
this changes the definition of `axes` to be a generator (and hence lazy) so that it would not spawn all 3 figure windows.  Instead first figure opens and draws, then after closing it the second one opens and draws, etc.  It's a fix for users who don't have `plt.ion()`.